From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: pvasilek <58226330+pvasilek@users.noreply.github.com>
Date: Sun, 29 Jun 2025 14:08:35 +0300
Subject: [PATCH] 8301: Decompiler: Fix div by zero on local variables retyping

Fix div by zero on local variables retyping

Div by zero occured if changing local variable type to greater size type in decompiler.
---
 .../java/ghidra/program/model/listing/VariableUtilities.java    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/listing/VariableUtilities.java b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/listing/VariableUtilities.java
index 043c0e7d43..a2ceb29c8d 100644
--- a/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/listing/VariableUtilities.java
+++ b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/listing/VariableUtilities.java
@@ -561,7 +561,7 @@ public class VariableUtilities {
 			callingConvention = compilerSpec.getDefaultCallingConvention();
 		}
 		int stackAlign = callingConvention.getStackParameterAlignment();
-		if (stackAlign < 0) {
+		if (stackAlign <= 0) {
 			stackAlign = 1;
 		}
 		int bias = 0;
-- 
2.45.1

